components/ActionSheet/src/private/MDCActionSheetListViewController.h

```diff
@@ -0,0 +1,77 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+#import "../MDCActionSheetController.h"
+#import "MDCActionSheetItemView.h"
+
+@interface MDCActionSheetListViewController : UITableViewController
```

Comment:
I think you can remove the need for this subclass if you extract the UITableViewDataSource into a model object. Essentially:

```
class ActionsModel: UITableViewDataSource {
  var actions: [Action]
}

// Within MDCActionSheetController
let model = ActionsModel() // Stored as an instance variable. Also is where all actions get stored
let tableViewController = UITableViewController()
tableViewController.dataSource = model
```

Reply:
Happy to discuss this in person if you have questions about the pattern.

Reply:
Done